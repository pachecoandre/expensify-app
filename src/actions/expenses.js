import uuid from 'uuid'

// ADD_EXPENSE ACTION
export const addExpense = (
   {
      description = '',
      note = '',
      amount = 0,
      createdAt = 0 
   } = {}
) => ({
   type: 'ADD_EXPENSE',
   expense: {
      id: uuid(),
      description,
      note,
      amount,
      createdAt
   }
})

// REMOVE_EXPENSE ACTION
export const removeExpense = ({ id }) => ({
   type: 'REMOVE_EXPENSE',
   id: id
})

// EDIT_EXPENSE ACTION
export const editExpense = (id, updates) => ({
   type: 'EDIT_EXPENSE',
   id,
   updates
})