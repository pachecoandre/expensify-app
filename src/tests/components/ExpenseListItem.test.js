import React from 'react'
import { shallow } from 'enzyme'
import ExpenseListItem from '../../components/ExpenseListItem'
import expenses from '../fixtures/expenses'

const expense = expenses[0]

test('should render ExpenseListItem with the expense provided', () => {
   const wrapper = shallow(<ExpenseListItem {...expense}/>)
   expect(wrapper).toMatchSnapshot()
})