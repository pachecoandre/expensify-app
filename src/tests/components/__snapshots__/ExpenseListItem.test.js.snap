// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem with the expense provided 1`] = `
<div>
  <[object Object]
    to="/edit/1"
  >
    <h3>
      Gum
    </h3>
  </[object Object]>
  <p>
    $ 1.95
    -
    31/Dec/1969
  </p>
</div>
`;
